### PR TITLE
Scikit-Learn Writing Algorithm

### PR DESCRIPTION
**Pull request by LouisSzeto**
#### Description
Add tutorial for `Scikit-Learn` library on [Writing Algorithm/Machine Learning](https://github.com/QuantConnect/Documentation/tree/master/02%20Writing%20Algorithms/30%20Machine%20Learning/03%20Popular%20Libraries).

#### Motivation and Context
Easy implementation example for users.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [X] New feature (non-breaking change which adds functionality)
- [X] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [X] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/02 Import Libraries.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<p>Import the <code>sklearn</code>, and <code>joblib</code> libraries.</p>
```
Comment (DerekMelchin) [inline 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/02 Import Libraries.html:1]: Remove `,`

File: 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/03 Create Subscriptions.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<p>In the <code>Initialize</code> method, subscribe to some data so you can train the <code>sklearn</code> model and make predictions.</p>
```
Comment (DerekMelchin) [inline 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/03 Create Subscriptions.html:1]: Add a link to "subscribe to some data" like we have in the keras tutorial

File: 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/04 Build Models.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<p>In this example, build a support vector regression prediction model using the following features and labels:</p>
```
Comment (DerekMelchin) [inline 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/04 Build Models.html:1]: "using" => "that uses"

File: 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/04 Build Models.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            <td>Labels</td>
+            <td>Daily percent return of the SPY over the next day</td>
+        </tr>
+    </tbody>
+</table>
+
+<p>The following image shows the time difference between the features and labels:</p>
+<img class="docs-image" src="https://cdn.quantconnect.com/i/tu/ml-keras-function.png">
+
+<p>Follow these steps to create a method to build the model:</p>
```
Comment (DerekMelchin) [inline 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/04 Build Models.html:25]: Condense these two steps into 1 line of code.

Then change the "Follow these steps ... " to "To build the model, call the GridSearchCV ..."

File: 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/06 Predict Labels.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<p>To predict the labels of new data, in the <code>OnData</code> method, get the most recent set of features and then call the <code>predict</code> method.</p>
+<div class="section-example-container">
+    <pre class="python">features, _ = self.get_features_and_labels()
+prediction = self.model.predict(new_X)
```
Comment (DerekMelchin) [inline 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/06 Predict Labels.html:4]: Instead of making predictions for all of the samples, shouldn't we just use the most recent one?

File: 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/07 Save Models.html [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<p>Follow these steps to save <code>sklearn</code> models into the <a href="https://www.quantconnect.com/docs/v2/writing-algorithms/object-store">ObjectStore</a>:</p>
+
+<ol>
+    <li>Set the key name of the model to be stored in the ObjectStore.</li>
```
Comment (DerekMelchin) [inline 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/07 Save Models.html:4]: > Set the key name you want to store the model under in the ObjectStore.

To match keras

File: 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/08 Load Models.html [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<p>You can load and trade with pre-trained <code>sklearn</code> models that you saved in the ObjectStore. To load a <code>sklearn</code> model from the ObjectStore, in the <code>Initialize</code> method, get the file path to the saved model and then call the <code>load</code> method.</p>
+<div class="section-example-container">
+    <pre class="python">def Initialize(self) -&gt; None:
+    if self.ObjectStore.ContainsKey(model_key):
+        file_name = self.ObjectStore.GetFilePath(model_key)
+        self.model = joblib.load(file_name)</pre>
+</div>
```
Comment (DerekMelchin) [inline 02 Writing Algorithms/30 Machine Learning/03 Popular Libraries/03 Scikit-Learn/08 Load Models.html:7]: After this line add the text from the keras tutorial that says "The ContainsKey method returns ..."

**Review by DerekMelchin — APPROVED** [review]
